### PR TITLE
#275 regulatory outline

### PR DESCRIPTION
**Pull request by zakiya**
Notes/questions:

- [x] When an issue is from another project, `#275` in commit doesn't work. 
- [x] Can/should we handle the lint/prettier pipeline in one command for all file types? (comparing to components/menu)
- [x] How can this be tested?
- [x] Is there a preferred templating engine that can be used to create markup and pull in data in this repo?
- [x] Are there css naming conventions? Preferred units?

### CODE REVIEW

**Comment by aaronhans** [comment]
Great questions!

We have a few spacing variables here https://github.com/cagov/design-system/blob/main/components/base-css/src/_variables.scss#L23 we should expand this set as we need more values so we can use standard spacing everywhere

Thanks for creating a separate template html file. I have a ticket to get that dynamically written into readmes so the markup isn't duplicated. We haven't setup our component gallery viewing system yet either yet but we can use the template HTML file there too.

The linting and prettier formatting commands and run hooks will get applied to the design-system repo as a whole. Right now there are still a lot of violations so need to fix them all before moving those commands outside of individual components.

When we write styles in our web components they are encapsulated by the custom element tag name which has a cagov- prefix.

When we create classnames that are applied outside of a custom element we can use the cagov- prefix to make them more unique like cagov-regulatory-outline

We have used https://every-layout.dev/ techniques in our design components already. I like how the W3C design system uses https://cube.fyi/ and think we should try to follow those patterns where they apply. That methodology isn't too prescriptive about class naming but helpful with organization. What do you think about CUBE CSS?

**Comment by zakiya** [comment]
@aaronhans 

1) Thanks for the reference to the custom properties in `_variables.scss`.  I tried testing that out for for regulatory outline on cannabis and I think the format for the spacing syntax needs to be  `--margin-sm: calc(var(--spacing-base) * 6);`

      Should we assume the user is including @cagov/base-css when designing components?

2) Got it.

3) Will do. 

4) Got it. 

5) Will do. 

6) 
      a)  Looks like every-layout.dev requires a payment. Will follow up about that on a different thread.  
      b) I'm not familiar with CUBE but reading through the docs it makes sense. I like using utility classes and BEM is familiar.

**Comment by aaronhans** [comment]
1. base-css is going to be there in all our instances but shouldn't be a hard dependency. When we've used values defined in there before we reference the css variable in the separate component and define a fallback value like https://github.com/cagov/design-system/blob/main/components/pagination/src/css/index.scss#L15

Then reference any referenced CSS variables in the variables section of the readme

6. Thanks for checking this stuff out. I'll ask Chad to get the team access to every-layout. The W3C design system referenced that in their documentation along with CUBE CSS. Koji and I like how that approach tries to solve some responsive layout problems without media queries so it is worth reviewing.

**Review by aaronhans — APPROVED** [review]
Excellent, thanks Zakiya! 

Love the tests! Cool use of counter-reset, counter-increment. I didn't know about those.

File: components/regulatory-outline/src/index.scss [hunk 1]
```diff
@@ -0,0 +1,43 @@
+cagov-regulatory-outline {
```
[the remark this answers is not in the document]
Comment (zakiya) [reply components/regulatory-outline/src/index.scss:1]: Ahh. I thought the pattern was to use custom element wrappers regardless. It'll be easier to more this for cmses using classes.

**Comment by zakiya** [comment]
Ready for review.

**Pull request merged**